### DISCUSSION TITLE
Speed overlay in kph

### DISCUSSION THREAD

**Discussion by kouyoumin** [gopro/labs#531] [Q&A] [answered]
When [adding an overlay to video](https://gopro.github.io/labs/control/overlays/)
GPS53 represents (ground) speed in m/s.
Is it possible to do some operations to the value, say, multiply by 3.6, to convert unit to km/h?

**Comment by dnewman-gpsw** [comment, marked as answer]
I was going to say this wasn't possible, but I've worked out a hack upon a hack, to make this work on HERO11.

The overlay's string message doesn't support math in its custom formatting, however math is support in the new scripting.  
Using the new visual scripting tool:
![image](https://github.com/gopro/labs/assets/10856084/b91ab643-ba0d-4c35-91ac-005ffc0438c8)

I created this:
![GPSSpeedDemoQR](https://github.com/gopro/labs/assets/10856084/af73aba9-261c-4ca6-ad0d-43b2e4f4a8ea)

It works in the sample video: https://www.dropbox.com/scl/fi/pvn6ee0h5d204r3sfbrma/GPSSpeedDemo.mp4?rlkey=zedmn6ngb5olbdcxatx5z7zcg&dl=0

This explains what the command is doing
![image](https://github.com/gopro/labs/assets/10856084/65dea8ff-338a-4bd4-babf-345bea7e668a)

The downside is it flashes "Overlays Active" continuously, as this command was never design to be scripted like this.  I will put a bug report in to see if it can be fixed.

**Reply by kouyoumin** [reply]
That's beyond my imagination. Thanks a lot.

**Comment by balls5000** [comment]
@dnewman-gpsw I have created this github account to thank you for your dashcam mod. I have 2 GoPros in my car and they work now better than ever (for the purpose of a dashcam).

This speed overlay in km/h could have been such a wonderful addition, but unfortunately the cameras turn off after approximately 2 minutes when using it, and then rebooting.

I have added a when reboot to the above mentioned script, to make it more permanent, but had to remove it all together because it froze and rebooted. 

Do you by any chance have a clue of how to fix this?

**Reply by dnewman-gpsw** [reply]
Using the overlays requires a lot of CPU power. To reduce that CPU power you need to reduce the size of the overlay. In that example above running an 80 point font was likely causing of too much CPU load, so make the size smaller and it should be more stable.

**Reply by balls5000** [reply]
Thanks @dnewman-gpsw  for your reply, after experimenting with it some further, I reached the following conclusions (hero11):

Manually scanning the QR code works every time

![Manual scan works-4_6_2024, 20_13_20](https://github.com/gopro/labs/assets/166044965/e948223e-2341-4151-bc8d-5cc8e82f6b29)

And it looks like this eventually 
![vlcsnap-2024-04-06-20h10m50s314](https://github.com/gopro/labs/assets/166044965/03f0a43b-b323-4248-9bf9-d6137190215c)

But trying to use your script in the "boot with", doesn't work even when the parameters are identical. It will just freeze after a few minutes and reboot.

![Doesn't work-4_6_2024, 20_16_30](https://github.com/gopro/labs/assets/166044965/5bc53f2f-4290-4a80-92e4-8fe783673810)

<img width="602" alt="Screenshot 2024-04-06 202343" src="https://github.com/gopro/labs/assets/166044965/06e820e7-20ec-4ca0-b1a7-69c3ba22ed8e">

But of course, this is after all, a hack you implemented after there was no stock support for displaying km/h, maybe a future version of labs will support displaying km/h? or I might be doing something wrong with blockly. By the way, as a precaution I tried decreasing the overlay size to 0.20 instead of 0.8, and the refresh ms to 100 instead of 10 (no idea if that actually helps)

All the best

**Reply by dnewman-gpsw** [reply]
oMDSLP=100 is not relevant is speed of the loop.  If you want to reduce the demand, add !N to the loop (this is half second pause), otherwise this will run as fast as it can.  However, I tried your boot command as is, and it worked without issue for me (I tested 4K30 at 160Mbit/s.)

**Comment by balls5000** [comment]
Thanks again for replying, it really means a lot. Well, in that case, my GoPro 11 hero might be the problem, as it struggles at 1080p 60. But then again, if manually scanning the QR and not applying at boot, it does work. Thanks again and God bless.

**Reply by dnewman-gpsw** [reply]
Try 30fps, just in case.

**Comment by iNNerKr** [comment]
Can you please tell me How do I remove the overlay completely? I am getting ISO: and SHUT: overlays with the number for these. And it's being recorded in video. Yes I am using GoPro 10 with this lab software... sorry I can't find an answer to this anywhere... this topics is about overlays so i thought you guys will be kind enough to let me know what to change where .. thanks 🙏

**Reply by dnewman-gpsw** [reply]
!RESET to remove everything or !MBURN="" to remove any permanent overlays.

**Reply by iNNerKr** [reply]
Thanks DNewman, I did that but it still shows ISO: and SHUT: info... plus in option of dropdown There is Burn not Mburn option?

**Reply by iNNerKr** [reply]
i think i tried many options... now nothing shows up, now every time i connect hdmi wire it goes in library mode and displays the list of files instead of live camera on laptop.

**Reply by dnewman-gpsw** [reply]
I think you talking about different things, you should have posted on a now clearly unrelated thread.  HDMI and overlays is not the same as Labs overlays.  Try ```!MHDMI=1```.   If that fails create a new thread with a clear description of what your problem is.

**Reply by kojinmalia** [reply]
Hi everyone, I'm also wanting to show km/h value but these qr codes do no work on my Hero 9. Is this just not possible or..?

**Reply by dnewman-gpsw** [reply]
HERO9 predates these advancements, MPH only on HERO9.

**Comment by kojinmalia** [comment]
It's available from hero 10, correct?

**Reply by dnewman-gpsw** [reply]
The newer scripting should work on HERO10. Seems to work, although I'm only at my desk so the overlay was 0 km/h.

**Reply by kojinmalia** [reply]
If I shows up then it should work yes. Thanks for trying. If you can reply my other post and test dual overlay (text + logo), would be great, thanks.
